### ISSUE TITLE
Pilot tests failing

### ISSUE THREAD

**Issue by gnirodi** [istio/istio#1968]
Beginning to notice too many failures for pilot tests. Can we find out how submits are allowed to go through. These don't seem to indicate any flakiness.

```
//pilot/adapter/config/crd:go_default_test
//pilot/platform/kube/admit:go_default_test 
//pilot/platform/kube:go_default_test 
//pilot/platform/kube/inject:go_default_test


=======================================
make test
INFO: Found 48 targets and 19 test targets...
FAIL: //pilot/platform/kube/admit:go_default_test (see /home/gnirodi/.cache/bazel/_bazel_gnirodi/ff0f802412b3a7e1f95314f131820562/execroot/io_istio_istio/bazel-out/local-fastbuild/testlogs/pilot/platform/kube/admit/go_default_test/test.log).
INFO: From Testing //pilot/platform/kube/admit:go_default_test:
==================== Test output for //pilot/platform/kube/admit:go_default_test:
--- FAIL: TestGetAPIServerExtensionCACert (0.44s)
	admit_test.go:487: GetAPIServerExtensionCACert() failed: Get https://35.197.14.76/api/v1/namespaces/kube-system/configmaps/extension-apiserver-authentication: error executing access token command "/usr/lib/google-cloud-sdk/bin/gcloud config config-helper --format=json": err=exit status 1 output=
FAIL
================================================================================
FAIL: //pilot/platform/kube:go_default_test (see /home/gnirodi/.cache/bazel/_bazel_gnirodi/ff0f802412b3a7e1f95314f131820562/execroot/io_istio_istio/bazel-out/local-fastbuild/testlogs/pilot/platform/kube/go_default_test/test.log).
INFO: From Testing //pilot/platform/kube:go_default_test:
==================== Test output for //pilot/platform/kube:go_default_test:
--- FAIL: TestServices (0.44s)
	controller_test.go:81: Post https://35.197.14.76/api/v1/namespaces: error executing access token command "/usr/lib/google-cloud-sdk/bin/gcloud config config-helper --format=json": err=exit status 1 output=
FAIL
================================================================================
FAIL: //pilot/platform/kube/inject:go_default_test (see /home/gnirodi/.cache/bazel/_bazel_gnirodi/ff0f802412b3a7e1f95314f131820562/execroot/io_istio_istio/bazel-out/local-fastbuild/testlogs/pilot/platform/kube/inject/go_default_test/test.log).
INFO: From Testing //pilot/platform/kube/inject:go_default_test:
==================== Test output for //pilot/platform/kube/inject:go_default_test:
E1203 23:33:18.238194      13 http.go:106] http: Server closed
--- FAIL: TestInitializerRun (0.41s)
	initializer_test.go:66: Post https://35.197.14.76/api/v1/namespaces: error executing access token command "/usr/lib/google-cloud-sdk/bin/gcloud config config-helper --format=json": err=exit status 1 output=
--- FAIL: TestInitialize (0.44s)
	initializer_test.go:88: Post https://35.197.14.76/api/v1/namespaces: error executing access token command "/usr/lib/google-cloud-sdk/bin/gcloud config config-helper --format=json": err=exit status 1 output=
--- FAIL: TestGetInitializerConfig (0.45s)
	inject_test.go:409: Post https://35.197.14.76/api/v1/namespaces: error executing access token command "/usr/lib/google-cloud-sdk/bin/gcloud config config-helper --format=json": err=exit status 1 output=
--- FAIL: TestGetMeshConfig (0.50s)
	inject_test.go:320: Post https://35.197.14.76/api/v1/namespaces: error executing access token command "/usr/lib/google-cloud-sdk/bin/gcloud config config-helper --format=json": err=exit status 1 output=
FAIL
================================================================================
FAIL: //pilot/adapter/config/crd:go_default_test (see /home/gnirodi/.cache/bazel/_bazel_gnirodi/ff0f802412b3a7e1f95314f131820562/execroot/io_istio_istio/bazel-out/local-fastbuild/testlogs/pilot/adapter/config/crd/go_default_test/test.log).
INFO: From Testing //pilot/adapter/config/crd:go_default_test:
==================== Test output for //pilot/adapter/config/crd:go_default_test:
--- FAIL: TestStoreInvariant (0.25s)
	client_test.go:72: Post https://35.197.14.76/api/v1/namespaces: error executing access token command "/usr/lib/google-cloud-sdk/bin/gcloud config config-helper --format=json": err=exit status 1 output=
--- FAIL: TestIstioConfig (0.25s)
	client_test.go:72: Post https://35.197.14.76/api/v1/namespaces: error executing access token command "/usr/lib/google-cloud-sdk/bin/gcloud config config-helper --format=json": err=exit status 1 output=
--- FAIL: TestControllerEvents (0.25s)
	client_test.go:72: Post https://35.197.14.76/api/v1/namespaces: error executing access token command "/usr/lib/google-cloud-sdk/bin/gcloud config config-helper --format=json": err=exit status 1 output=
--- FAIL: TestControllerCacheFreshness (0.24s)
	client_test.go:72: Post https://35.197.14.76/api/v1/namespaces: error executing access token command "/usr/lib/google-cloud-sdk/bin/gcloud config config-helper --format=json": err=exit status 1 output=
--- FAIL: TestControllerClientSync (0.24s)
	client_test.go:72: Post https://35.197.14.76/api/v1/namespaces: error executing access token command "/usr/lib/google-cloud-sdk/bin/gcloud config config-helper --format=json": err=exit status 1 output=
FAIL
================================================================================
INFO: Elapsed time: 35.730s, Critical Path: 30.83s
//pilot/adapter/config/aggregate:go_default_xtest                        PASSED in 0.4s
//pilot/adapter/config/crd:go_default_xtest                              PASSED in 0.9s
//pilot/adapter/config/file:go_default_xtest                             PASSED in 13.9s
//pilot/adapter/config/ingress:go_default_test                           PASSED in 30.8s
//pilot/adapter/config/memory:go_default_xtest                           PASSED in 0.5s
//pilot/adapter/serviceregistry/aggregate:go_default_test                PASSED in 0.2s
//pilot/model:go_default_test                                            PASSED in 0.1s
//pilot/model:go_default_xtest                                           PASSED in 0.3s
//pilot/platform/consul:go_default_test                                  PASSED in 0.5s
//pilot/platform/eureka:go_default_test                                  PASSED in 0.4s
//pilot/proxy/envoy:go_default_test                                      PASSED in 3.2s
//pilot/proxy:go_default_test                                            PASSED in 0.7s
//pilot/proxy:go_default_xtest                                           PASSED in 0.4s
//pilot/test/mock:go_default_test                                        PASSED in 0.3s
//pilot/tools/version:go_default_test                                    PASSED in 0.1s
//pilot/adapter/config/crd:go_default_test                               FAILED in 1.5s
  /home/gnirodi/.cache/bazel/_bazel_gnirodi/ff0f802412b3a7e1f95314f131820562/execroot/io_istio_istio/bazel-out/local-fastbuild/testlogs/pilot/adapter/config/crd/go_default_test/test.log
//pilot/platform/kube/admit:go_default_test                              FAILED in 0.8s
  /home/gnirodi/.cache/bazel/_bazel_gnirodi/ff0f802412b3a7e1f95314f131820562/execroot/io_istio_istio/bazel-out/local-fastbuild/testlogs/pilot/platform/kube/admit/go_default_test/test.log
//pilot/platform/kube:go_default_test                                    FAILED in 0.8s
  /home/gnirodi/.cache/bazel/_bazel_gnirodi/ff0f802412b3a7e1f95314f131820562/execroot/io_istio_istio/bazel-out/local-fastbuild/testlogs/pilot/platform/kube/go_default_test/test.log
//pilot/platform/kube/inject:go_default_test                             FAILED in 1.9s
  /home/gnirodi/.cache/bazel/_bazel_gnirodi/ff0f802412b3a7e1f95314f131820562/execroot/io_istio_istio/bazel-out/local-fastbuild/testlogs/pilot/platform/kube/inject/go_default_test/test.log

Executed 19 out of 19 tests: 15 tests pass and 4 fail locally.
make: *** [test] Error 3
```

**Comment by vadimeisenbergibm** [comment]
all the tests pass on minikube for me.

**Comment by gnirodi** [comment]
Downgrading to P2 given that I was able to recreate my entire dev setup just to get things going :-(

For posterity sake, here were the issues and updating docs may save others some grief:

1. My docker daemon seems to be starting before the network service. At the time, ipv4 forwarding is disabled. So it was unable to go get any URL that was external while building the container
This was resolved by: 
   sudo /etc/init.d/docker restart
This may be applicable only for ubuntu dev environments

2. Seems like my cluster had RBAC setup for the wrong user. The error message "error executing access token command"  is not at all helpful. It should have said "The K8s cluster needs an RBAC for user XXX and point to the playbook"
I resolved this by running:
kubectl create clusterrolebinding cluster-admin-binding --clusterrole=cluster-admin --user="$(gcloud config get-value core/account)"
